widget/embedded/src/pages/SettingsPage.tsx

```diff
@@ -1,93 +1,238 @@
   const { navigateBackFrom } = useNavigateBack();
 
   const infiniteApprove = useSettingsStore.use.infiniteApprove();
   const toggleInfiniteApprove = useSettingsStore.use.toggleInfiniteApprove();
   const loadingMetaStatus = useMetaStore.use.loadingStatus();
 
-  const uniqueSwappersGroups: Array<{
-    title: string;
-    logo: string;
-    type: 'BRIDGE' | 'AGGREGATOR' | 'DEX';
-    selected: boolean;
-  }> = [];
-  removeDuplicateFrom(swappers.map((s) => s.swapperGroup))
-    .map((swapperGroup) => {
-      return swappers.find((s) => s.swapperGroup === swapperGroup);
-    })
-    .find((s) => {
-      if (s) {
-        for (const type of s.types) {
-          uniqueSwappersGroups.push({
-            title: s.swapperGroup,
-            logo: s.logo,
-            type,
-            selected: !disabledLiquiditySources.includes(s.swapperGroup),
-          });
-        }
-      }
-    });
-  supportedSwappers &&
-    uniqueSwappersGroups.filter(
-      (item) => supportedSwappers.filter((s) => s === item.title).length > 0
-    );
-
-  const supportedUniqueSwappersGroups = supportedSwappers
-    ? uniqueSwappersGroups.filter(
-        (item) => supportedSwappers.filter((s) => s === item.title).length > 0
-      )
-    : uniqueSwappersGroups;
+  const supportedUniqueSwappersGroups =
+    getUniqueSwappersGroups(supportedSwappers);
+
+  const bridgeSources = supportedUniqueSwappersGroups.filter(
+    (uniqueItem) =>
+      uniqueItem.type === 'BRIDGE' || uniqueItem.type === 'AGGREGATOR'
+  );
+
+  const selectedBridgeSources = bridgeSources.filter(
+    (uniqueItem) => uniqueItem.selected
+  );
+
+  const exchangeSources = supportedUniqueSwappersGroups.filter(
+    (uniqueItem) => uniqueItem.type === 'DEX'
+  );
+
+  const selectedExchangeSources = exchangeSources.filter(
+    (uniqueItem) => uniqueItem.selected
+  );
+
+  const bridgeItem = {
+    id: '1',
+    title: <SettingItemTitle title="Enable bridges" />,
+    end: (
+      <>
+        {loadingMetaStatus === 'success' && (
+          <Typography variant="body" size="medium" color="neutral900">
+            {bridgeSources.length !== selectedBridgeSources.length
+              ? `${selectedBridgeSources.length} / ${bridgeSources.length}`
+              : bridgeSources.length}
+          </Typography>
+        )}
+        {loadingMetaStatus === 'failed' && (
+          <Typography variant="body" size="medium" color="$error500">
+            Loading failed
+          </Typography>
+        )}
+        <Divider direction="horizontal" size={8} />
+        <ChevronRightIcon color="gray" />
+      </>
+    ),
+    onClick: () => navigate(navigationRoutes.bridges),
+  };
+
+  const exchangeItem = {
+    id: '2',
+    title: <SettingItemTitle title="Enable exchanges" />,
+    end: (
+      <>
+        {loadingMetaStatus === 'success' && (
+          <Typography variant="body" size="medium" color="neutral900">
+            {exchangeSources.length !== selectedExchangeSources.length
```

Comment:
Please put them on a variable.
`const totalExchanges= ...`
and 
`const totalEnabledExchanges = ...`.

Apply same thing for `bridges` section.